core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -421,6 +421,13 @@ object SparkHadoopUtil {
 
   val SPARK_YARN_CREDS_COUNTER_DELIM = "-"
 
+  // Just load HdfsConfiguration into the class loader to add
+  // hdfs-site.xml as a default configuration file otherwise
+  // some HDFS related configurations doesn't ship to Executors and
+  // it can cause UnknownHostException when NameNode HA is enabled.
+  // See SPARK-11227 for more details.
+  Utils.classForName("org.apache.hadoop.hdfs.HdfsConfiguration")
```

Comment:
actually I'd prefer not to reference hdfs-site.xml directly.  HdfsConfiguration should know what it needs to load.   For instance it also loads the defaults.  HdfsConfiguration is also marked as @InterfaceAudience.Private so ideally we shouldn't be using it directly.   Based on my other comments I would like to understand better why this isn't loaded on driver already.